fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanAccrualActivityProcessingServiceImpl.java

```diff
@@ -0,0 +1,96 @@
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.fineract.portfolio.loanaccount.domain.Loan;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanRepaymentScheduleInstallment;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanRepositoryWrapper;
+import org.apache.fineract.portfolio.loanaccount.domain.LoanTransaction;
+import org.apache.fineract.portfolio.loanaccount.exception.LoanNotFoundException;
+import org.springframework.stereotype.Component;
+
+@Slf4j
+@Component
+@RequiredArgsConstructor
+public class LoanAccrualActivityProcessingServiceImpl implements LoanAccrualActivityProcessingService {
+
+    private final LoanRepositoryWrapper loanRepositoryWrapper;
+    private final LoanReadPlatformService loanReadPlatformService;
+    private final LoanWritePlatformService loanWritePlatformService;
+
+    public void makeAccrualActivityTransaction(Long accountId, final LocalDate currentDate) {
+        if (accountId != null) {
+            Loan loan = this.loanRepositoryWrapper.findOneWithNotFoundDetection(accountId, true);
+            if (loan == null) {
+                throw new LoanNotFoundException(accountId);
+            }
+            makeAccrualActivityTransaction(loan, currentDate);
+        }
+    }
+
+    public void makeAccrualActivityTransaction(Loan loan, final LocalDate currentDate) {
+        if (loan.getLoanProductRelatedDetail().isEnableAccrualActivityPosting()) {
+            // check if loan has installment due on business day
+            Optional<LoanRepaymentScheduleInstallment> first = loan.getRepaymentScheduleInstallments().stream()
+                    .filter(loanRepaymentScheduleInstallment -> loanRepaymentScheduleInstallment.getDueDate().isEqual(currentDate))
+                    .findFirst();
+            if (first.isPresent()) {
+                final LoanRepaymentScheduleInstallment installment = first.get();
+                // check if there is no not-replayed-accrual-activity related to business date
+                List<LoanTransaction> loanTransactions = loan.getLoanTransactions(loanTransaction -> loanTransaction.isNotReversed()
+                        && loanTransaction.isAccrualActivity() && loanTransaction.getTransactionDate().isEqual(currentDate));
+                if (loanTransactions.isEmpty()) {
+                    loanWritePlatformService.makeAccrualActivityTransaction(loan, installment, currentDate);
+                }
+            }
+        }
+    }
+
+    @Override
+    public List<Throwable> findAndPostAccrualActivities(LocalDate yesterday) {
+        List<Throwable> errors = new ArrayList<>();
+
+        Long offsetId = 0L;
+        List<Long> loanAccounts;
+        int limit = 100;
+        do {
+            loanAccounts = loanReadPlatformService.fetchLoanIdsForAccrualActivityPosting(yesterday, offsetId, limit);
```

Comment:
I think Pageable would be better there... also we dont need to use the LoanReadPlatformService. I would prefer if we just simply create a Repository class which filter based on the params (including pageable) and just go through on that.

Example: `ExternalAssetOwnerJournalEntryMappingRepository.findByExternalOwnerId`